go-controller/pkg/network-attach-def-controller/network_attach_def_controller.go

```diff
@@ -280,3 +301,90 @@ func nadNeedsUpdate(oldNAD, newNAD *nettypes.NetworkAttachmentDefinition) bool {
 
 	return !reflect.DeepEqual(oldNAD.Spec, newNAD.Spec)
 }
+
+func (nadController *NetAttachDefinitionController) unsetActiveNetworkForNamespace(namespace string, oldNetwork util.NetInfo) {
+	nadController.primaryNetworks.LockKey(namespace)
+	defer nadController.primaryNetworks.UnlockKey(namespace)
+	primaryNetworks, loaded := nadController.primaryNetworks.Load(namespace)
+	if loaded {
+		if len(oldNetwork.GetNADs()) > 0 {
+			// some NAD still references this network, do not remove it,
+			// but still update the syncmap with the latest version removing the NAD
+			// reference
+			for name := range primaryNetworks {
+				if name == oldNetwork.GetNetworkName() {
+					primaryNetworks[name] = oldNetwork
+					nadController.primaryNetworks.Store(namespace, primaryNetworks)
+					return
+				}
+			}
+		} else if len(primaryNetworks) > 1 {
+			// multiple primary networks for this namespace, just remove the one
+			delete(primaryNetworks, oldNetwork.GetNetworkName())
+			nadController.primaryNetworks.Store(namespace, primaryNetworks)
+		} else {
+			// only 1 network, delete the entire namespace from being tracked
+			nadController.primaryNetworks.Delete(namespace)
+		}
+	}
+}
+
+func (nadController *NetAttachDefinitionController) setActiveNetworkForNamespace(namespace string, ensureNetwork util.NetInfo) {
+	nadController.primaryNetworks.LockKey(namespace)
+	defer nadController.primaryNetworks.UnlockKey(namespace)
+	var primaryNetInfos map[string]util.NetInfo
+	var loaded bool
+	if primaryNetInfos, loaded = nadController.primaryNetworks.Load(namespace); loaded {
+		primaryNetInfos[ensureNetwork.GetNetworkName()] = ensureNetwork
+	} else {
+		primaryNetInfos = map[string]util.NetInfo{ensureNetwork.GetNetworkName(): ensureNetwork}
+	}
+	nadController.primaryNetworks.Store(namespace, primaryNetInfos)
+}
+
+func (nadController *NetAttachDefinitionController) GetActiveNetworkForNamespace(namespace string) (util.NetInfo, error) {
```

Comment:
So the `NetInfo` that we return here, wouldn't it just have all the NADs for the network instead of just eh NADs in the namespace? That's what the old `GetActiveNetworkForNamespace` was doing.

Reply:
That's a good point. I'm returning all NADs associated with this network, rather than just the one scoped to the namespace (note there can be only one returned, cause otherwise we return the error for multiple primary NADs). I'll fix this.